lighthouse-core/gather/gather-runner.js

```diff
@@ -129,19 +178,38 @@ class GatherRunner {
     const status = {msg: 'Initializing…', id: 'lh:gather:setupDriver'};
     log.time(status);
     const resetStorage = !options.settings.disableStorageReset;
-    await driver.assertNoSameOriginServiceWorkerClients(options.requestedUrl);
+    const session = driver.defaultSession;
+
+    // Assert no service workers are still installed, so we test that they would actually be installed for a new user.
+    // TODO(FR-COMPAT): re-evaluate the necessity of this check
+    await GatherRunner.assertNoSameOriginServiceWorkerClients(session, options.requestedUrl);
+
+    // Emulate our target device screen and user agent.
     await emulation.emulate(driver.defaultSession, options.settings);
-    await driver.enableRuntimeEvents();
-    await driver.enableAsyncStacks();
+
+    // Enable `Debugger` domain to receive async stacktrace information on network request initiators.
+    // This is critical for tracing certain performance simulation situations.
+    session.on('Debugger.paused', () => session.sendCommand('Debugger.resume'));
```

Comment:
this will be extracted along with the other shared driver setup in a future PR that focuses on gather-runner